src/utilities/sql/SqlFile_Impl.cpp

```diff
@@ -2504,18 +2504,29 @@ namespace openstudio{
 
           unsigned month = sqlite3_column_int(sqlStmtPtr, b++);
           unsigned day = sqlite3_column_int(sqlStmtPtr, b++);
-          unsigned intervalMinutes = sqlite3_column_int(sqlStmtPtr, b++); // used for run periods
-          if (reportingFrequency == ReportingFrequency::Annual){
-            intervalMinutes = 8760; // used for annual periods
+
+          // In cases where you report the same meter key for eg at Daily and at Timestep frequency
+          // the intervalMinutes will be reported by E+ for the Timestep one, so you get the wrong one for Daily...
+          // And since we can compute this easily, might as well do it
+          unsigned intervalMinutes;
+          if (reportingFrequency == ReportingFrequency::Hourly) {
+            intervalMinutes = 60;
+          } else if (reportingFrequency == ReportingFrequency::Daily) {
+            intervalMinutes = 24 * 60;
+          } else if (reportingFrequency == ReportingFrequency::Monthly) {
+            intervalMinutes = day * 24 * 60;
+          } else {
```

Comment:
For fixed-length reporting frequencies, just calculate it instead of querying the Sql which could be wrong.